### PR TITLE
Check for root early

### PR DESCRIPTION
**Pull request by mdhorn**
Change the timing where we check for the root user.

Fixes: 171

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -456,7 +456,7 @@ func listBlockDevices(userDefined []*BlockDevice) ([]*BlockDevice, error) {
 		return nil, err
 	}
 
-	if err = utils.VerifyRootUser(); err == nil {
+	if errString := utils.VerifyRootUser(); errString == "" {
```
Comment (dorileo) [inline storage/storage.go]: With the early check, do we need this check at all?

Comment (mdhorn) [reply storage/storage.go]: Maybe not? I was concerned there was some other code flow which used storage before the first check. You think I should drop it?

Comment (dorileo) [reply storage/storage.go]: Yep, please drop it.

Comment (mdhorn) [reply storage/storage.go]: Dropping. First fall out is one of the storage tests is now failing when not run as root. Updating...

Comment (mdhorn) [reply storage/storage.go]: @dorileo Now now the coverage numbers of storage have dropped. :(

Comment (mdhorn) [reply storage/storage.go]: @dorileo Updates made as we discussed offline.
New changes are pushed and testing now.

**Pull request merged**